tests/utils/gamestate_utils.cc

```diff
@@ -34,6 +39,21 @@ std::unique_ptr<GameState> AdvanceUntilState(std::unique_ptr<GameState> state,
   return state;
 }
 
+std::unique_ptr<GameState> AdvanceThroughState(std::unique_ptr<GameState> state,
+                                               StateFunctionType targetState,
+                                               int maxIterations) {
+  // First advance until we reach the target state
+  state = AdvanceUntilState(std::move(state), targetState, maxIterations);
+
+  // Then advance once more through that state
+  state = AdvanceGameState(std::move(state));
+
+  // Verify we made it the target state
+  EXPECT_EQ(state->currState, targetState);
```

Comment:
I would keep this in the test code. I think EXPECTs outside of a test function can have weird interactions. We would throw anyways if we didn't reach the expected state in maxIterations.

Reply:
I still wanted some sort of assertions around this working so I built out a separate test file!